Use slf4j and set logging level to INFO for jaegertracing classes

pavolloffay opened the pull request:
Fixes #15 

Example output:
```
docker run  --env="STORAGE=cassandra" jaegertracing/spark-dependencies:latest | fpaste                 11:11 
17/10/10 09:11:55 INFO CassandraDependenciesJob: Running Dependencies job for 2017-10-10T00:00Z: 1507593600000000 ≤ Span.timestamp 1507679999999999
17/10/10 09:11:55 INFO SparkContext: Running Spark version 2.1.1
17/10/10 09:11:55 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/10/10 09:11:55 INFO SecurityManager: Changing view acls to: root
17/10/10 09:11:55 INFO SecurityManager: Changing modify acls to: root
17/10/10 09:11:55 INFO SecurityManager: Changing view acls groups to: 
17/10/10 09:11:55 INFO SecurityManager: Changing modify acls groups to: 
17/10/10 09:11:55 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(root); groups with view permissions: Set(); users  with modify permissions: Set(root); groups with modify permissions: Set()
17/10/10 09:11:56 INFO Utils: Successfully started service 'sparkDriver' on port 35969.
17/10/10 09:11:56 INFO SparkEnv: Registering MapOutputTracker
17/10/10 09:11:56 INFO SparkEnv: Registering BlockManagerMaster
17/10/10 09:11:56 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
17/10/10 09:11:56 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
17/10/10 09:11:56 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-61add854-62c0-420e-a38a-30bfd980d925
17/10/10 09:11:56 INFO MemoryStore: MemoryStore started with capacity 3.9 GB
17/10/10 09:11:56 INFO SparkEnv: Registering OutputCommitCoordinator
17/10/10 09:11:56 INFO SparkContext: Added JAR /app/jaeger-spark-dependencies/target/jaeger-spark-dependencies-0.0.1-SNAPSHOT.jar at spark://172.17.0.2:35969/jars/jaeger-spark-dependencies-0.0.1-SNAPSHOT.jar with timestamp 1507626716216
17/10/10 09:11:56 INFO Executor: Starting executor ID driver on host localhost
17/10/10 09:11:56 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 38633.
17/10/10 09:11:56 INFO NettyBlockTransferService: Server created on 172.17.0.2:38633
17/10/10 09:11:56 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
17/10/10 09:11:56 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 172.17.0.2, 38633, None)
17/10/10 09:11:56 INFO BlockManagerMasterEndpoint: Registering block manager 172.17.0.2:38633 with 3.9 GB RAM, BlockManagerId(driver, 172.17.0.2, 38633, None)
17/10/10 09:11:56 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 172.17.0.2, 38633, None)
17/10/10 09:11:56 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 172.17.0.2, 38633, None)
17/10/10 09:11:56 INFO log: Logging initialized @1503ms
17/10/10 09:11:57 INFO Native: Could not load JNR C Library, native system calls through this library will not be available (set this logger level to DEBUG to see the full stack trace).
17/10/10 09:11:57 INFO ClockFactory: Using java.lang.System clock to generate timestamps.
17/10/10 09:11:57 INFO NettyUtil: Found Netty's native epoll transport in the classpath, using it
17/10/10 09:12:00 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
17/10/10 09:12:00 INFO MemoryStore: MemoryStore cleared
17/10/10 09:12:00 INFO BlockManager: BlockManager stopped
17/10/10 09:12:00 INFO BlockManagerMaster: BlockManagerMaster stopped
17/10/10 09:12:00 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
17/10/10 09:12:00 INFO SparkContext: Successfully stopped SparkContext
Exception in thread "main" java.io.IOException: Failed to open native connection to Cassandra at {127.0.0.1}:9042
```

pavolloffay commented:
> Whereas the "Done" entry added by this PR isn't on the sample output (Done, %d dependency objects created)

It's not because the job failed, I run it without C*, but you can see the first log when the job started.

jpkrohling commented:
> It's not because the job failed, I run it without C*, but you can see the first log when the job started.

Then it's missing a clear message about the job not being successful and the cause for the failure :)

pavolloffay commented:
> Then it's missing a clear message about the job not being successful and the cause for the failure :) 

because I didn't copy it all :) 

Here is the full log for cassandra:
```
docker run --rm -it   --env="STORAGE=cassandra" --env="CASSANDRA_KEYSPACE=jaeger_v1_test" --net=host --env="SPARK_LOCAL_IP=127.0.0.1" jaegertracing/spark-dependencies:latest
17/10/10 09:53:25 INFO CassandraDependenciesJob: Running Dependencies job for 2017-10-10T00:00Z: 1507593600000000 ≤ Span.timestamp 1507679999999999
17/10/10 09:53:25 INFO SparkContext: Running Spark version 2.1.1
17/10/10 09:53:25 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/10/10 09:53:26 INFO SecurityManager: Changing view acls to: root
17/10/10 09:53:26 INFO SecurityManager: Changing modify acls to: root
17/10/10 09:53:26 INFO SecurityManager: Changing view acls groups to: 
17/10/10 09:53:26 INFO SecurityManager: Changing modify acls groups to: 
17/10/10 09:53:26 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(root); groups with view permissions: Set(); users  with modify permissions: Set(root); groups with modify permissions: Set()
17/10/10 09:53:26 INFO Utils: Successfully started service 'sparkDriver' on port 34407.
17/10/10 09:53:26 INFO SparkEnv: Registering MapOutputTracker
17/10/10 09:53:26 INFO SparkEnv: Registering BlockManagerMaster
17/10/10 09:53:26 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
17/10/10 09:53:26 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
17/10/10 09:53:26 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-74c72cfa-23ba-49d1-bf95-9a1e8bb00270
17/10/10 09:53:26 INFO MemoryStore: MemoryStore started with capacity 3.9 GB
17/10/10 09:53:26 INFO SparkEnv: Registering OutputCommitCoordinator
17/10/10 09:53:26 INFO SparkContext: Added JAR /app/jaeger-spark-dependencies/target/jaeger-spark-dependencies-0.0.1-SNAPSHOT.jar at spark://127.0.0.1:34407/jars/jaeger-spark-dependencies-0.0.1-SNAPSHOT.jar with timestamp 1507629206546
17/10/10 09:53:26 INFO Executor: Starting executor ID driver on host localhost
17/10/10 09:53:26 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 44157.
17/10/10 09:53:26 INFO NettyBlockTransferService: Server created on 127.0.0.1:44157
17/10/10 09:53:26 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
17/10/10 09:53:26 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 127.0.0.1, 44157, None)
17/10/10 09:53:26 INFO BlockManagerMasterEndpoint: Registering block manager 127.0.0.1:44157 with 3.9 GB RAM, BlockManagerId(driver, 127.0.0.1, 44157, None)
17/10/10 09:53:26 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 127.0.0.1, 44157, None)
17/10/10 09:53:26 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 127.0.0.1, 44157, None)
17/10/10 09:53:26 INFO log: Logging initialized @1658ms
17/10/10 09:53:28 INFO Native: Could not load JNR C Library, native system calls through this library will not be available (set this logger level to DEBUG to see the full stack trace).
17/10/10 09:53:28 INFO ClockFactory: Using java.lang.System clock to generate timestamps.
17/10/10 09:53:28 INFO NettyUtil: Found Netty's native epoll transport in the classpath, using it
17/10/10 09:53:28 INFO Cluster: New Cassandra host localhost/127.0.0.1:9042 added
17/10/10 09:53:28 INFO CassandraConnector: Connected to Cassandra cluster: test
17/10/10 09:53:29 INFO SparkContext: Starting job: collect at DependenciesSparkHelper.java:42
17/10/10 09:53:29 INFO DAGScheduler: Registering RDD 2 (mapToPair at CassandraDependenciesJob.java:156)
17/10/10 09:53:29 INFO DAGScheduler: Registering RDD 7 (mapToPair at DependenciesSparkHelper.java:39)
17/10/10 09:53:29 INFO DAGScheduler: Got job 0 (collect at DependenciesSparkHelper.java:42) with 17 output partitions
17/10/10 09:53:29 INFO DAGScheduler: Final stage: ResultStage 2 (collect at DependenciesSparkHelper.java:42)
17/10/10 09:53:29 INFO DAGScheduler: Parents of final stage: List(ShuffleMapStage 1)
17/10/10 09:53:29 INFO DAGScheduler: Missing parents: List(ShuffleMapStage 1)
17/10/10 09:53:29 INFO DAGScheduler: Submitting ShuffleMapStage 0 (MapPartitionsRDD[2] at mapToPair at CassandraDependenciesJob.java:156), which has no missing parents
17/10/10 09:53:29 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 16.0 KB, free 3.9 GB)
17/10/10 09:53:29 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 7.7 KB, free 3.9 GB)
17/10/10 09:53:29 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 127.0.0.1:44157 (size: 7.7 KB, free: 3.9 GB)
17/10/10 09:53:29 INFO SparkContext: Created broadcast 0 from broadcast at DAGScheduler.scala:996
17/10/10 09:53:29 INFO DAGScheduler: Submitting 17 missing tasks from ShuffleMapStage 0 (MapPartitionsRDD[2] at mapToPair at CassandraDependenciesJob.java:156)
17/10/10 09:53:29 INFO TaskSchedulerImpl: Adding task set 0.0 with 17 tasks
17/10/10 09:53:29 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0, localhost, executor driver, partition 0, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 1.0 in stage 0.0 (TID 1, localhost, executor driver, partition 1, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 2.0 in stage 0.0 (TID 2, localhost, executor driver, partition 2, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 3.0 in stage 0.0 (TID 3, localhost, executor driver, partition 3, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 4.0 in stage 0.0 (TID 4, localhost, executor driver, partition 4, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 5.0 in stage 0.0 (TID 5, localhost, executor driver, partition 5, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 6.0 in stage 0.0 (TID 6, localhost, executor driver, partition 6, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 7.0 in stage 0.0 (TID 7, localhost, executor driver, partition 7, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO Executor: Running task 0.0 in stage 0.0 (TID 0)
17/10/10 09:53:29 INFO Executor: Running task 2.0 in stage 0.0 (TID 2)
17/10/10 09:53:29 INFO Executor: Running task 3.0 in stage 0.0 (TID 3)
17/10/10 09:53:29 INFO Executor: Running task 1.0 in stage 0.0 (TID 1)
17/10/10 09:53:29 INFO Executor: Running task 4.0 in stage 0.0 (TID 4)
17/10/10 09:53:29 INFO Executor: Running task 5.0 in stage 0.0 (TID 5)
17/10/10 09:53:29 INFO Executor: Running task 6.0 in stage 0.0 (TID 6)
17/10/10 09:53:29 INFO Executor: Running task 7.0 in stage 0.0 (TID 7)
17/10/10 09:53:29 INFO Executor: Fetching spark://127.0.0.1:34407/jars/jaeger-spark-dependencies-0.0.1-SNAPSHOT.jar with timestamp 1507629206546
17/10/10 09:53:29 INFO TransportClientFactory: Successfully created connection to /127.0.0.1:34407 after 19 ms (0 ms spent in bootstraps)
17/10/10 09:53:29 INFO Utils: Fetching spark://127.0.0.1:34407/jars/jaeger-spark-dependencies-0.0.1-SNAPSHOT.jar to /tmp/spark-a8965e28-05cd-46b7-9aee-b025cdad7e22/userFiles-18ead614-c4b1-40ac-b262-3e5ac1638284/fetchFileTemp2574977266672933564.tmp
17/10/10 09:53:29 INFO Executor: Adding file:/tmp/spark-a8965e28-05cd-46b7-9aee-b025cdad7e22/userFiles-18ead614-c4b1-40ac-b262-3e5ac1638284/jaeger-spark-dependencies-0.0.1-SNAPSHOT.jar to class loader
17/10/10 09:53:29 INFO Executor: Finished task 7.0 in stage 0.0 (TID 7). 1156 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 5.0 in stage 0.0 (TID 5). 1156 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 6.0 in stage 0.0 (TID 6). 1156 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 0.0 in stage 0.0 (TID 0). 1069 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 2.0 in stage 0.0 (TID 2). 1156 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 4.0 in stage 0.0 (TID 4). 1156 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 1.0 in stage 0.0 (TID 1). 1156 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 3.0 in stage 0.0 (TID 3). 1156 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Starting task 8.0 in stage 0.0 (TID 8, localhost, executor driver, partition 8, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO Executor: Running task 8.0 in stage 0.0 (TID 8)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 9.0 in stage 0.0 (TID 9, localhost, executor driver, partition 9, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO Executor: Running task 9.0 in stage 0.0 (TID 9)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 10.0 in stage 0.0 (TID 10, localhost, executor driver, partition 10, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO Executor: Running task 10.0 in stage 0.0 (TID 10)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 11.0 in stage 0.0 (TID 11, localhost, executor driver, partition 11, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 12.0 in stage 0.0 (TID 12, localhost, executor driver, partition 12, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO Executor: Running task 11.0 in stage 0.0 (TID 11)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 7.0 in stage 0.0 (TID 7) in 411 ms on localhost (executor driver) (1/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 2.0 in stage 0.0 (TID 2) in 428 ms on localhost (executor driver) (2/17)
17/10/10 09:53:29 INFO Executor: Running task 12.0 in stage 0.0 (TID 12)
17/10/10 09:53:29 INFO Executor: Finished task 9.0 in stage 0.0 (TID 9). 996 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 8.0 in stage 0.0 (TID 8). 1083 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Finished task 6.0 in stage 0.0 (TID 6) in 422 ms on localhost (executor driver) (3/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 0.0 in stage 0.0 (TID 0) in 474 ms on localhost (executor driver) (4/17)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 13.0 in stage 0.0 (TID 13, localhost, executor driver, partition 13, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO Executor: Running task 13.0 in stage 0.0 (TID 13)
17/10/10 09:53:29 INFO Executor: Finished task 10.0 in stage 0.0 (TID 10). 996 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Starting task 14.0 in stage 0.0 (TID 14, localhost, executor driver, partition 14, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO Executor: Running task 14.0 in stage 0.0 (TID 14)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 15.0 in stage 0.0 (TID 15, localhost, executor driver, partition 15, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO Executor: Running task 15.0 in stage 0.0 (TID 15)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 16.0 in stage 0.0 (TID 16, localhost, executor driver, partition 16, NODE_LOCAL, 7568 bytes)
17/10/10 09:53:29 INFO Executor: Running task 16.0 in stage 0.0 (TID 16)
17/10/10 09:53:29 INFO Executor: Finished task 12.0 in stage 0.0 (TID 12). 1083 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 11.0 in stage 0.0 (TID 11). 996 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Finished task 5.0 in stage 0.0 (TID 5) in 446 ms on localhost (executor driver) (5/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 9.0 in stage 0.0 (TID 9) in 45 ms on localhost (executor driver) (6/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 1.0 in stage 0.0 (TID 1) in 458 ms on localhost (executor driver) (7/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 4.0 in stage 0.0 (TID 4) in 454 ms on localhost (executor driver) (8/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 11.0 in stage 0.0 (TID 11) in 39 ms on localhost (executor driver) (9/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 3.0 in stage 0.0 (TID 3) in 464 ms on localhost (executor driver) (10/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 10.0 in stage 0.0 (TID 10) in 57 ms on localhost (executor driver) (11/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 8.0 in stage 0.0 (TID 8) in 64 ms on localhost (executor driver) (12/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 12.0 in stage 0.0 (TID 12) in 51 ms on localhost (executor driver) (13/17)
17/10/10 09:53:29 INFO Executor: Finished task 16.0 in stage 0.0 (TID 16). 996 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 13.0 in stage 0.0 (TID 13). 1083 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Finished task 16.0 in stage 0.0 (TID 16) in 31 ms on localhost (executor driver) (14/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 13.0 in stage 0.0 (TID 13) in 39 ms on localhost (executor driver) (15/17)
17/10/10 09:53:29 INFO Executor: Finished task 14.0 in stage 0.0 (TID 14). 996 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 15.0 in stage 0.0 (TID 15). 996 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Finished task 14.0 in stage 0.0 (TID 14) in 46 ms on localhost (executor driver) (16/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 15.0 in stage 0.0 (TID 15) in 46 ms on localhost (executor driver) (17/17)
17/10/10 09:53:29 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool 
17/10/10 09:53:29 INFO DAGScheduler: ShuffleMapStage 0 (mapToPair at CassandraDependenciesJob.java:156) finished in 0.531 s
17/10/10 09:53:29 INFO DAGScheduler: looking for newly runnable stages
17/10/10 09:53:29 INFO DAGScheduler: running: Set()
17/10/10 09:53:29 INFO DAGScheduler: waiting: Set(ShuffleMapStage 1, ResultStage 2)
17/10/10 09:53:29 INFO DAGScheduler: failed: Set()
17/10/10 09:53:29 INFO DAGScheduler: Submitting ShuffleMapStage 1 (MapPartitionsRDD[7] at mapToPair at DependenciesSparkHelper.java:39), which has no missing parents
17/10/10 09:53:29 INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 18.3 KB, free 3.9 GB)
17/10/10 09:53:29 INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 8.6 KB, free 3.9 GB)
17/10/10 09:53:29 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 127.0.0.1:44157 (size: 8.6 KB, free: 3.9 GB)
17/10/10 09:53:29 INFO SparkContext: Created broadcast 1 from broadcast at DAGScheduler.scala:996
17/10/10 09:53:29 INFO DAGScheduler: Submitting 17 missing tasks from ShuffleMapStage 1 (MapPartitionsRDD[7] at mapToPair at DependenciesSparkHelper.java:39)
17/10/10 09:53:29 INFO TaskSchedulerImpl: Adding task set 1.0 with 17 tasks
17/10/10 09:53:29 INFO TaskSetManager: Starting task 0.0 in stage 1.0 (TID 17, localhost, executor driver, partition 0, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 1.0 in stage 1.0 (TID 18, localhost, executor driver, partition 1, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 2.0 in stage 1.0 (TID 19, localhost, executor driver, partition 2, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 3.0 in stage 1.0 (TID 20, localhost, executor driver, partition 3, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 4.0 in stage 1.0 (TID 21, localhost, executor driver, partition 4, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 5.0 in stage 1.0 (TID 22, localhost, executor driver, partition 5, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 6.0 in stage 1.0 (TID 23, localhost, executor driver, partition 6, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 7.0 in stage 1.0 (TID 24, localhost, executor driver, partition 7, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO Executor: Running task 0.0 in stage 1.0 (TID 17)
17/10/10 09:53:29 INFO Executor: Running task 1.0 in stage 1.0 (TID 18)
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO Executor: Running task 2.0 in stage 1.0 (TID 19)
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 3 ms
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 2 ms
17/10/10 09:53:29 INFO Executor: Running task 3.0 in stage 1.0 (TID 20)
17/10/10 09:53:29 INFO Executor: Running task 4.0 in stage 1.0 (TID 21)
17/10/10 09:53:29 INFO Executor: Running task 7.0 in stage 1.0 (TID 24)
17/10/10 09:53:29 INFO Executor: Running task 6.0 in stage 1.0 (TID 23)
17/10/10 09:53:29 INFO Executor: Running task 5.0 in stage 1.0 (TID 22)
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 9 ms
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 2 ms
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 11 ms
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 7 ms
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 5 ms
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 2 ms
17/10/10 09:53:29 INFO Executor: Finished task 7.0 in stage 1.0 (TID 24). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 5.0 in stage 1.0 (TID 22). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 2.0 in stage 1.0 (TID 19). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Starting task 8.0 in stage 1.0 (TID 25, localhost, executor driver, partition 8, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO Executor: Finished task 0.0 in stage 1.0 (TID 17). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Starting task 9.0 in stage 1.0 (TID 26, localhost, executor driver, partition 9, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 7.0 in stage 1.0 (TID 24) in 62 ms on localhost (executor driver) (1/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 0.0 in stage 1.0 (TID 17) in 73 ms on localhost (executor driver) (2/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 5.0 in stage 1.0 (TID 22) in 65 ms on localhost (executor driver) (3/17)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 10.0 in stage 1.0 (TID 27, localhost, executor driver, partition 10, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO Executor: Finished task 1.0 in stage 1.0 (TID 18). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 3.0 in stage 1.0 (TID 20). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 6.0 in stage 1.0 (TID 23). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Running task 8.0 in stage 1.0 (TID 25)
17/10/10 09:53:29 INFO Executor: Running task 9.0 in stage 1.0 (TID 26)
17/10/10 09:53:29 INFO Executor: Running task 10.0 in stage 1.0 (TID 27)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 11.0 in stage 1.0 (TID 28, localhost, executor driver, partition 11, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 12.0 in stage 1.0 (TID 29, localhost, executor driver, partition 12, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 13.0 in stage 1.0 (TID 30, localhost, executor driver, partition 13, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO Executor: Running task 13.0 in stage 1.0 (TID 30)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 6.0 in stage 1.0 (TID 23) in 76 ms on localhost (executor driver) (4/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 2.0 in stage 1.0 (TID 19) in 84 ms on localhost (executor driver) (5/17)
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 1 ms
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 1 ms
17/10/10 09:53:29 INFO Executor: Finished task 9.0 in stage 1.0 (TID 26). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Finished task 13.0 in stage 1.0 (TID 30). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Running task 12.0 in stage 1.0 (TID 29)
17/10/10 09:53:29 INFO Executor: Finished task 4.0 in stage 1.0 (TID 21). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO Executor: Running task 11.0 in stage 1.0 (TID 28)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 3.0 in stage 1.0 (TID 20) in 82 ms on localhost (executor driver) (6/17)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 14.0 in stage 1.0 (TID 31, localhost, executor driver, partition 14, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 9.0 in stage 1.0 (TID 26) in 25 ms on localhost (executor driver) (7/17)
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 7 ms
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:29 INFO Executor: Finished task 10.0 in stage 1.0 (TID 27). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 1 ms
17/10/10 09:53:29 INFO Executor: Finished task 8.0 in stage 1.0 (TID 25). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Starting task 15.0 in stage 1.0 (TID 32, localhost, executor driver, partition 15, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO Executor: Running task 14.0 in stage 1.0 (TID 31)
17/10/10 09:53:29 INFO TaskSetManager: Starting task 16.0 in stage 1.0 (TID 33, localhost, executor driver, partition 16, PROCESS_LOCAL, 5822 bytes)
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 6 ms
17/10/10 09:53:29 INFO Executor: Running task 16.0 in stage 1.0 (TID 33)
17/10/10 09:53:29 INFO Executor: Running task 15.0 in stage 1.0 (TID 32)
17/10/10 09:53:29 INFO Executor: Finished task 11.0 in stage 1.0 (TID 28). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Finished task 1.0 in stage 1.0 (TID 18) in 106 ms on localhost (executor driver) (8/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 8.0 in stage 1.0 (TID 25) in 42 ms on localhost (executor driver) (9/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 11.0 in stage 1.0 (TID 28) in 34 ms on localhost (executor driver) (10/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 13.0 in stage 1.0 (TID 30) in 24 ms on localhost (executor driver) (11/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 10.0 in stage 1.0 (TID 27) in 36 ms on localhost (executor driver) (12/17)
17/10/10 09:53:29 INFO TaskSetManager: Finished task 4.0 in stage 1.0 (TID 21) in 102 ms on localhost (executor driver) (13/17)
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:29 INFO Executor: Finished task 14.0 in stage 1.0 (TID 31). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Finished task 14.0 in stage 1.0 (TID 31) in 28 ms on localhost (executor driver) (14/17)
17/10/10 09:53:29 INFO Executor: Finished task 12.0 in stage 1.0 (TID 29). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Finished task 12.0 in stage 1.0 (TID 29) in 41 ms on localhost (executor driver) (15/17)
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:29 INFO Executor: Finished task 16.0 in stage 1.0 (TID 33). 1967 bytes result sent to driver
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:29 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 1 ms
17/10/10 09:53:29 INFO TaskSetManager: Finished task 16.0 in stage 1.0 (TID 33) in 28 ms on localhost (executor driver) (16/17)
17/10/10 09:53:29 INFO Executor: Finished task 15.0 in stage 1.0 (TID 32). 1880 bytes result sent to driver
17/10/10 09:53:29 INFO TaskSetManager: Finished task 15.0 in stage 1.0 (TID 32) in 35 ms on localhost (executor driver) (17/17)
17/10/10 09:53:29 INFO TaskSchedulerImpl: Removed TaskSet 1.0, whose tasks have all completed, from pool 
17/10/10 09:53:30 INFO DAGScheduler: ShuffleMapStage 1 (mapToPair at DependenciesSparkHelper.java:39) finished in 0.138 s
17/10/10 09:53:30 INFO DAGScheduler: looking for newly runnable stages
17/10/10 09:53:30 INFO DAGScheduler: running: Set()
17/10/10 09:53:30 INFO DAGScheduler: waiting: Set(ResultStage 2)
17/10/10 09:53:30 INFO DAGScheduler: failed: Set()
17/10/10 09:53:30 INFO DAGScheduler: Submitting ResultStage 2 (MapPartitionsRDD[9] at values at DependenciesSparkHelper.java:41), which has no missing parents
17/10/10 09:53:30 INFO MemoryStore: Block broadcast_2 stored as values in memory (estimated size 4.3 KB, free 3.9 GB)
17/10/10 09:53:30 INFO MemoryStore: Block broadcast_2_piece0 stored as bytes in memory (estimated size 2.4 KB, free 3.9 GB)
17/10/10 09:53:30 INFO BlockManagerInfo: Added broadcast_2_piece0 in memory on 127.0.0.1:44157 (size: 2.4 KB, free: 3.9 GB)
17/10/10 09:53:30 INFO SparkContext: Created broadcast 2 from broadcast at DAGScheduler.scala:996
17/10/10 09:53:30 INFO DAGScheduler: Submitting 17 missing tasks from ResultStage 2 (MapPartitionsRDD[9] at values at DependenciesSparkHelper.java:41)
17/10/10 09:53:30 INFO TaskSchedulerImpl: Adding task set 2.0 with 17 tasks
17/10/10 09:53:30 INFO TaskSetManager: Starting task 0.0 in stage 2.0 (TID 34, localhost, executor driver, partition 0, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 1.0 in stage 2.0 (TID 35, localhost, executor driver, partition 1, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 2.0 in stage 2.0 (TID 36, localhost, executor driver, partition 2, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 3.0 in stage 2.0 (TID 37, localhost, executor driver, partition 3, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 4.0 in stage 2.0 (TID 38, localhost, executor driver, partition 4, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 5.0 in stage 2.0 (TID 39, localhost, executor driver, partition 5, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 6.0 in stage 2.0 (TID 40, localhost, executor driver, partition 6, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 7.0 in stage 2.0 (TID 41, localhost, executor driver, partition 7, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO Executor: Running task 0.0 in stage 2.0 (TID 34)
17/10/10 09:53:30 INFO Executor: Running task 2.0 in stage 2.0 (TID 36)
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO Executor: Running task 3.0 in stage 2.0 (TID 37)
17/10/10 09:53:30 INFO Executor: Running task 1.0 in stage 2.0 (TID 35)
17/10/10 09:53:30 INFO Executor: Running task 7.0 in stage 2.0 (TID 41)
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 1 ms
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO Executor: Running task 6.0 in stage 2.0 (TID 40)
17/10/10 09:53:30 INFO Executor: Running task 4.0 in stage 2.0 (TID 38)
17/10/10 09:53:30 INFO Executor: Finished task 7.0 in stage 2.0 (TID 41). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Finished task 3.0 in stage 2.0 (TID 37). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO Executor: Running task 5.0 in stage 2.0 (TID 39)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 8.0 in stage 2.0 (TID 42, localhost, executor driver, partition 8, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 7.0 in stage 2.0 (TID 41) in 16 ms on localhost (executor driver) (1/17)
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO Executor: Finished task 4.0 in stage 2.0 (TID 38). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO Executor: Finished task 0.0 in stage 2.0 (TID 34). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 1 ms
17/10/10 09:53:30 INFO TaskSetManager: Starting task 9.0 in stage 2.0 (TID 43, localhost, executor driver, partition 9, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 3.0 in stage 2.0 (TID 37) in 22 ms on localhost (executor driver) (2/17)
17/10/10 09:53:30 INFO Executor: Running task 9.0 in stage 2.0 (TID 43)
17/10/10 09:53:30 INFO Executor: Running task 8.0 in stage 2.0 (TID 42)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 10.0 in stage 2.0 (TID 44, localhost, executor driver, partition 10, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 11.0 in stage 2.0 (TID 45, localhost, executor driver, partition 11, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 0.0 in stage 2.0 (TID 34) in 29 ms on localhost (executor driver) (3/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 4.0 in stage 2.0 (TID 38) in 24 ms on localhost (executor driver) (4/17)
17/10/10 09:53:30 INFO Executor: Running task 10.0 in stage 2.0 (TID 44)
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 1 ms
17/10/10 09:53:30 INFO Executor: Finished task 1.0 in stage 2.0 (TID 35). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Running task 11.0 in stage 2.0 (TID 45)
17/10/10 09:53:30 INFO Executor: Finished task 5.0 in stage 2.0 (TID 39). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO TaskSetManager: Starting task 12.0 in stage 2.0 (TID 46, localhost, executor driver, partition 12, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO Executor: Running task 12.0 in stage 2.0 (TID 46)
17/10/10 09:53:30 INFO Executor: Finished task 6.0 in stage 2.0 (TID 40). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO TaskSetManager: Starting task 13.0 in stage 2.0 (TID 47, localhost, executor driver, partition 13, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 19 ms
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO Executor: Finished task 9.0 in stage 2.0 (TID 43). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 11 ms
17/10/10 09:53:30 INFO Executor: Running task 13.0 in stage 2.0 (TID 47)
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 1 ms
17/10/10 09:53:30 INFO TaskSetManager: Starting task 14.0 in stage 2.0 (TID 48, localhost, executor driver, partition 14, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO Executor: Finished task 8.0 in stage 2.0 (TID 42). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO TaskSetManager: Finished task 9.0 in stage 2.0 (TID 43) in 26 ms on localhost (executor driver) (5/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 6.0 in stage 2.0 (TID 40) in 43 ms on localhost (executor driver) (6/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 1.0 in stage 2.0 (TID 35) in 47 ms on localhost (executor driver) (7/17)
17/10/10 09:53:30 INFO Executor: Finished task 10.0 in stage 2.0 (TID 44). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Finished task 12.0 in stage 2.0 (TID 46). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Finished task 2.0 in stage 2.0 (TID 36). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Running task 14.0 in stage 2.0 (TID 48)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 15.0 in stage 2.0 (TID 49, localhost, executor driver, partition 15, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 5.0 in stage 2.0 (TID 39) in 46 ms on localhost (executor driver) (8/17)
17/10/10 09:53:30 INFO Executor: Finished task 11.0 in stage 2.0 (TID 45). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Finished task 13.0 in stage 2.0 (TID 47). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO TaskSetManager: Starting task 16.0 in stage 2.0 (TID 50, localhost, executor driver, partition 16, PROCESS_LOCAL, 5833 bytes)
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO Executor: Running task 16.0 in stage 2.0 (TID 50)
17/10/10 09:53:30 INFO Executor: Running task 15.0 in stage 2.0 (TID 49)
17/10/10 09:53:30 INFO Executor: Finished task 14.0 in stage 2.0 (TID 48). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Getting 0 non-empty blocks out of 17 blocks
17/10/10 09:53:30 INFO ShuffleBlockFetcherIterator: Started 0 remote fetches in 0 ms
17/10/10 09:53:30 INFO Executor: Finished task 15.0 in stage 2.0 (TID 49). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO TaskSetManager: Finished task 8.0 in stage 2.0 (TID 42) in 39 ms on localhost (executor driver) (9/17)
17/10/10 09:53:30 INFO Executor: Finished task 16.0 in stage 2.0 (TID 50). 1722 bytes result sent to driver
17/10/10 09:53:30 INFO TaskSetManager: Finished task 12.0 in stage 2.0 (TID 46) in 21 ms on localhost (executor driver) (10/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 10.0 in stage 2.0 (TID 44) in 33 ms on localhost (executor driver) (11/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 2.0 in stage 2.0 (TID 36) in 56 ms on localhost (executor driver) (12/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 11.0 in stage 2.0 (TID 45) in 36 ms on localhost (executor driver) (13/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 13.0 in stage 2.0 (TID 47) in 23 ms on localhost (executor driver) (14/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 16.0 in stage 2.0 (TID 50) in 15 ms on localhost (executor driver) (15/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 15.0 in stage 2.0 (TID 49) in 17 ms on localhost (executor driver) (16/17)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 14.0 in stage 2.0 (TID 48) in 24 ms on localhost (executor driver) (17/17)
17/10/10 09:53:30 INFO TaskSchedulerImpl: Removed TaskSet 2.0, whose tasks have all completed, from pool 
17/10/10 09:53:30 INFO DAGScheduler: ResultStage 2 (collect at DependenciesSparkHelper.java:42) finished in 0.070 s
17/10/10 09:53:30 INFO DAGScheduler: Job 0 finished: collect at DependenciesSparkHelper.java:42, took 0.917018 s
17/10/10 09:53:30 INFO SparkContext: Starting job: runJob at RDDFunctions.scala:36
17/10/10 09:53:30 INFO DAGScheduler: Got job 1 (runJob at RDDFunctions.scala:36) with 8 output partitions
17/10/10 09:53:30 INFO DAGScheduler: Final stage: ResultStage 3 (runJob at RDDFunctions.scala:36)
17/10/10 09:53:30 INFO DAGScheduler: Parents of final stage: List()
17/10/10 09:53:30 INFO DAGScheduler: Missing parents: List()
17/10/10 09:53:30 INFO DAGScheduler: Submitting ResultStage 3 (ParallelCollectionRDD[10] at parallelize at CassandraDependenciesJob.java:169), which has no missing parents
17/10/10 09:53:30 INFO MemoryStore: Block broadcast_3 stored as values in memory (estimated size 12.9 KB, free 3.9 GB)
17/10/10 09:53:30 INFO MemoryStore: Block broadcast_3_piece0 stored as bytes in memory (estimated size 5.8 KB, free 3.9 GB)
17/10/10 09:53:30 INFO BlockManagerInfo: Added broadcast_3_piece0 in memory on 127.0.0.1:44157 (size: 5.8 KB, free: 3.9 GB)
17/10/10 09:53:30 INFO SparkContext: Created broadcast 3 from broadcast at DAGScheduler.scala:996
17/10/10 09:53:30 INFO DAGScheduler: Submitting 8 missing tasks from ResultStage 3 (ParallelCollectionRDD[10] at parallelize at CassandraDependenciesJob.java:169)
17/10/10 09:53:30 INFO TaskSchedulerImpl: Adding task set 3.0 with 8 tasks
17/10/10 09:53:30 INFO TaskSetManager: Starting task 0.0 in stage 3.0 (TID 51, localhost, executor driver, partition 0, PROCESS_LOCAL, 5955 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 1.0 in stage 3.0 (TID 52, localhost, executor driver, partition 1, PROCESS_LOCAL, 5955 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 2.0 in stage 3.0 (TID 53, localhost, executor driver, partition 2, PROCESS_LOCAL, 5955 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 3.0 in stage 3.0 (TID 54, localhost, executor driver, partition 3, PROCESS_LOCAL, 5955 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 4.0 in stage 3.0 (TID 55, localhost, executor driver, partition 4, PROCESS_LOCAL, 5955 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 5.0 in stage 3.0 (TID 56, localhost, executor driver, partition 5, PROCESS_LOCAL, 5955 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 6.0 in stage 3.0 (TID 57, localhost, executor driver, partition 6, PROCESS_LOCAL, 5955 bytes)
17/10/10 09:53:30 INFO TaskSetManager: Starting task 7.0 in stage 3.0 (TID 58, localhost, executor driver, partition 7, PROCESS_LOCAL, 7861 bytes)
17/10/10 09:53:30 INFO Executor: Running task 0.0 in stage 3.0 (TID 51)
17/10/10 09:53:30 INFO Executor: Running task 1.0 in stage 3.0 (TID 52)
17/10/10 09:53:30 INFO Executor: Running task 4.0 in stage 3.0 (TID 55)
17/10/10 09:53:30 INFO Executor: Running task 2.0 in stage 3.0 (TID 53)
17/10/10 09:53:30 INFO Executor: Running task 7.0 in stage 3.0 (TID 58)
17/10/10 09:53:30 INFO Executor: Running task 6.0 in stage 3.0 (TID 57)
17/10/10 09:53:30 INFO Executor: Running task 5.0 in stage 3.0 (TID 56)
17/10/10 09:53:30 INFO Executor: Running task 3.0 in stage 3.0 (TID 54)
17/10/10 09:53:30 INFO TableWriter: Wrote 0 rows to jaeger_v1_test.dependencies in 0.028 s.
17/10/10 09:53:30 INFO TableWriter: Wrote 0 rows to jaeger_v1_test.dependencies in 0.038 s.
17/10/10 09:53:30 INFO TableWriter: Wrote 0 rows to jaeger_v1_test.dependencies in 0.028 s.
17/10/10 09:53:30 INFO TableWriter: Wrote 0 rows to jaeger_v1_test.dependencies in 0.036 s.
17/10/10 09:53:30 INFO TableWriter: Wrote 0 rows to jaeger_v1_test.dependencies in 0.042 s.
17/10/10 09:53:30 INFO Executor: Finished task 5.0 in stage 3.0 (TID 56). 1012 bytes result sent to driver
17/10/10 09:53:30 INFO TableWriter: Wrote 0 rows to jaeger_v1_test.dependencies in 0.038 s.
17/10/10 09:53:30 INFO TableWriter: Wrote 0 rows to jaeger_v1_test.dependencies in 0.041 s.
17/10/10 09:53:30 INFO Executor: Finished task 3.0 in stage 3.0 (TID 54). 1012 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Finished task 6.0 in stage 3.0 (TID 57). 925 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Finished task 4.0 in stage 3.0 (TID 55). 925 bytes result sent to driver
17/10/10 09:53:30 INFO TaskSetManager: Finished task 5.0 in stage 3.0 (TID 56) in 74 ms on localhost (executor driver) (1/8)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 6.0 in stage 3.0 (TID 57) in 85 ms on localhost (executor driver) (2/8)
17/10/10 09:53:30 INFO Executor: Finished task 0.0 in stage 3.0 (TID 51). 1012 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Finished task 2.0 in stage 3.0 (TID 53). 925 bytes result sent to driver
17/10/10 09:53:30 INFO Executor: Finished task 1.0 in stage 3.0 (TID 52). 925 bytes result sent to driver
17/10/10 09:53:30 INFO TaskSetManager: Finished task 0.0 in stage 3.0 (TID 51) in 97 ms on localhost (executor driver) (3/8)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 2.0 in stage 3.0 (TID 53) in 92 ms on localhost (executor driver) (4/8)
17/10/10 09:53:30 INFO TableWriter: Wrote 1 rows to jaeger_v1_test.dependencies in 0.067 s.
17/10/10 09:53:30 INFO TaskSetManager: Finished task 1.0 in stage 3.0 (TID 52) in 94 ms on localhost (executor driver) (5/8)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 3.0 in stage 3.0 (TID 54) in 94 ms on localhost (executor driver) (6/8)
17/10/10 09:53:30 INFO TaskSetManager: Finished task 4.0 in stage 3.0 (TID 55) in 93 ms on localhost (executor driver) (7/8)
17/10/10 09:53:30 INFO Executor: Finished task 7.0 in stage 3.0 (TID 58). 1093 bytes result sent to driver
17/10/10 09:53:30 INFO TaskSetManager: Finished task 7.0 in stage 3.0 (TID 58) in 91 ms on localhost (executor driver) (8/8)
17/10/10 09:53:30 INFO TaskSchedulerImpl: Removed TaskSet 3.0, whose tasks have all completed, from pool 
17/10/10 09:53:30 INFO DAGScheduler: ResultStage 3 (runJob at RDDFunctions.scala:36) finished in 0.103 s
17/10/10 09:53:30 INFO DAGScheduler: Job 1 finished: runJob at RDDFunctions.scala:36, took 0.113376 s
17/10/10 09:53:30 INFO CassandraDependenciesJob: Done, 0 dependency objects created
17/10/10 09:53:30 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
17/10/10 09:53:30 INFO MemoryStore: MemoryStore cleared
17/10/10 09:53:30 INFO BlockManager: BlockManager stopped
17/10/10 09:53:30 INFO BlockManagerMaster: BlockManagerMaster stopped
17/10/10 09:53:30 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
17/10/10 09:53:30 INFO SparkContext: Successfully stopped SparkContext
17/10/10 09:53:37 INFO CassandraConnector: Disconnected from Cassandra cluster: test
17/10/10 09:53:38 INFO ShutdownHookManager: Shutdown hook called
17/10/10 09:53:38 INFO SerialShutdownHooks: Successfully executed shutdown hook: Clearing session cache for C* connector
17/10/10 09:53:38 INFO ShutdownHookManager: Deleting directory /tmp/spark-a8965e28-05cd-46b7-9aee-b025cdad7e22
```

jpkrohling commented:
If you say this log is clean enough to not confuse an admin and complete enough to allow an admin to know what went wrong when it fails, then it's OK to merge.

But if I were an admin and needed to debug a failure by looking at this log, I wouldn't know to even start looking for the failure... This log is big and full of information I don't need. If I *know* what I'm looking for (`Done, 0 dependency objects created`), I can find it, but otherwise, it's hard to know the status of this job.

For instance:

```
17/10/10 09:53:30 INFO DAGScheduler: Parents of final stage: List()
17/10/10 09:53:30 INFO DAGScheduler: Missing parents: List()
```

What is this about? Is this a problem? Is this expected? IMO, this belongs to "trace" level, not "info" (this specific entry is just an example, similar questions apply to all log entries here).

pavolloffay commented:
> What is this about? Is this a problem? Is this expected? IMO, this belongs to "trace" level, not "info" (this specific entry is just an example, similar questions apply to all log entries here).

perhaps we can ask in Apache Spark repo :). The info level in Spark is too verbose. 

I will revert it back to default WARN level and use INFO in our classes.

pavolloffay commented:
Now the log looks like this: 

```
STORAGE=cassandra CASSANDRA_KEYSPACE=jaeger_v1_test SPARK_LOCAL_IP=127.0.0.1 java -jar jaeger-spark-dependencies/target/jaeger-spark-dependencies-0.0.1-SNAPSHOT.jar
17/10/10 12:31:14 INFO CassandraDependenciesJob: Running Dependencies job for 2017-10-10T00:00Z: 1507593600000000 ≤ Span.timestamp 1507679999999999
17/10/10 12:31:14 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/10/10 12:31:18 INFO CassandraDependenciesJob: Done, 0 dependency objects created

```

jpkrohling commented:
Far better :) The date format could be improved to remove ambiguity (yyyy-mm-ddZhh:mm:ss.ssss would be my preference), and logging useful for debugging failures would be nice to have as well, but those can be added over time.

Being picky, here are a couple extra improvements:

> 17/10/10 12:31:14 INFO CassandraDependenciesJob: Running Dependencies job for 2017-10-10T00:00Z: 1507593600000000 ≤ Span.timestamp 1507679999999999

How about converting the timestamp into a human readable date?

> 17/10/10 12:31:14 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable

Is this something an admin has to worry about? Ideally, this would be out of the logs (or a native hadoop library would be in the image)

jpkrohling commented:
By the way: this PR LGTM already. The final parts of my last comment were "nice to have", but not required.

pavolloffay commented:
> Is this something an admin has to worry about? Ideally, this would be out of the logs (or a native hadoop library would be in the image)

I think it only warns if it runs in local mode. 


I will merge. Further improvements can be done as separate PRs.
